pkg/apis/intelligence/v1alpha1/types.go

```diff
@@ -31,19 +41,32 @@ type NetworkPolicyRecommendation struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Type                string      `json:"jobType,omitempty"`
-	Limit               int         `json:"limit,omitempty"`
-	PolicyType          string      `json:"policyType,omitempty"`
-	StartTime           metav1.Time `json:"startTime,omitempty"`
-	EndTime             metav1.Time `json:"endTime,omitempty"`
-	NSAllowList         []string    `json:"nsAllowList,omitempty"`
-	ExcludeLabels       bool        `json:"excludeLabels,omitempty"`
-	ToServices          bool        `json:"toServices,omitempty"`
-	ExecutorInstances   int         `json:"executorInstances,omitempty"`
-	DriverCoreRequest   string      `json:"driverCoreRequest,omitempty"`
-	DriverMemory        string      `json:"driverMemory,omitempty"`
-	ExecutorCoreRequest string      `json:"executorCoreRequest,omitempty"`
-	ExecutorMemory      string      `json:"executorMemory,omitempty"`
+	Type                JobType                           `json:"jobType,omitempty"`
+	Limit               int                               `json:"limit,omitempty"`
+	PolicyType          string                            `json:"policyType,omitempty"`
+	StartTime           metav1.Time                       `json:"startTime,omitempty"`
+	EndTime             metav1.Time                       `json:"endTime,omitempty"`
```

Comment:
Aa Salvatore suggested in Slack, I plan to rename these two fields at CRD side to `IntervalStart` and `IntervalEnd` to reduce the confusion that these two look like the start and end time of NetworkPolicyRecommendation CR. I'm not sure whether we should also update here or if you have other suggestions on the naming.

Reply:
the new naming sounds good to me. will also update doc.

Reply:
Or also StartInterval and EndInterval... really makes no difference to me, I prefer having field semantically similar close when reading them in alphabetical order, but other might prefer having "Start" and "End" at the beginning as it immediately provides detailed semantic info.